docs/src/gradient_accumulation.md

```diff
@@ -0,0 +1,132 @@
+# Gradient Accumulation
+
+Consider some function
+$$f(x) = g(x) + h(x)$$.
+If we would like the derivative of $f$ with respect to $x$ we must compute it for each part and then sum them, i.e. 
+$$\frac{\partial f}{\partial x} = \frac{\partial g}{\partial x} + \frac{\partial h}{\partial x}$$.
+In general, we must accumulate (sum) gradients from each sub-part of a program where a variable is used.
+
+
+Consider for example:
+```julia
+function sum_first_and_second(X::Array{Float64})
+    a = X[1]
+    b = X[2]
+    y = a + b
+    return y
```

Comment:
I have a feeling you want to be as explicit about this as possible, but

```suggestion
    return X[1] + X[2]
```

Reply:
I want to name `a`, `b` and `y`, so I can talk about them later.